tests/protocols/test_http.py

```diff
@@ -161,13 +172,17 @@ def add_done_callback(self, callback):
 @contextlib.contextmanager
 def get_connected_protocol(app, protocol_cls, event_loop, **kwargs):
     loop = MockLoop(event_loop)
+    asyncio._set_running_loop(loop)
     transport = MockTransport()
     config = Config(app=app, **kwargs)
     server_state = ServerState()
     protocol = protocol_cls(config=config, server_state=server_state, _loop=loop)
     protocol.connection_made(transport)
-    yield protocol
-    protocol.loop.close()
+    try:
+        yield protocol
+    finally:
+        protocol.loop.close()
+        asyncio._set_running_loop(None)
```

Comment:
I'm certainly not familiar enough with all asyncio internals but why the last line here, isn't it enough to close the loop ?
edit just tested removing it and it seems like tests pass fine without, but I'm sure you have an explanation ;)

Reply:
You have to detach the running loop otherwise get_running_loop() will continue to return this MockLoop